### ISSUE TITLE
Enabling/Disabling waiter's break checkbox

### ISSUE THREAD

**Issue by jlazzaridean** [usc-csci201-fall2013/restaurant#65]
I have a bizarre bug where the waiter's checkbox sometimes does not become re-enabled after the waiter's break response returns from the host. it works about 70% of the time, but it will randomly stay grayed out sometimes. i've checked through my code quite a bit and i should be re-enabling it every time. has anyone had a similar problem? @mcwhitak (i cleaned up my code from earlier)

**Comment by jlazzaridean** [comment]
and the act of creating this issue of course magically helped me find the bug. closed #65

**Comment by brianykim** [comment]
@jlazzaridean Can you share your insights? Having a very similar issue. I'm updating the panel and everything :/

**Comment by jlazzaridean** [comment]
I think the issue was a race condition one where it was sometimes re-enabling the checkbox before it was disabled. When I switched the order so that the line of code disabling it occurred before the line where I was calling the waiterGui (which called the waiter, and the host, and then returned back). Basically, I set it so that in a given step, it always disabled the checkbox before trying to change any states or check for any breaks.
